File: src/main/java/com/gmail/borlandlp/minigamesdtools/activepoints/type/block/FlatSquarePoint.java
package com.gmail.borlandlp.minigamesdtools.activepoints.type.block;

import com.gmail.borlandlp.minigamesdtools.activepoints.BuildSchema;
import com.gmail.borlandlp.minigamesdtools.util.GeometryHelper;
import com.gmail.borlandlp.minigamesdtools.util.geometry.GeneratedBlockSchema;
import org.bukkit.Location;
import org.bukkit.Material;

import java.util.HashMap;
import java.util.Map;

public class FlatSquarePoint extends PrimitiveBlockPoint {
    @Override
    public BuildSchema getBuildSchema() {
        GeneratedBlockSchema generatedBlockSchema = GeometryHelper.generateFlatSquare(this.getLocation(), this.getRadius(), this.isHollow());
        Map<Location, Material> blocks = new HashMap<>();
        generatedBlockSchema.getBorderBlocks().forEach(location -> blocks.put(location, this.getBorderMaterial()));
        generatedBlockSchema.getFillerBlocks().forEach(location -> blocks.put(location, this.getFillerMaterial()));

        return new BuildSchema(blocks);
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/activepoints/type/block/FlatSquarePoint.kt
package com.gmail.borlandlp.minigamesdtools.activepoints.type.block

import com.gmail.borlandlp.minigamesdtools.activepoints.BuildSchema
import com.gmail.borlandlp.minigamesdtools.util.GeometryHelper.generateFlatSquare
import org.bukkit.Location
import org.bukkit.Material
import java.util.*
import java.util.function.Consumer

class FlatSquarePoint : PrimitiveBlockPoint() {
    override val buildSchema: BuildSchema
        get() {
            val generatedBlockSchema =
                generateFlatSquare(location!!, radius, isHollow)
            val blocks: MutableMap<Location, Material> =
                HashMap()
            generatedBlockSchema.borderBlocks.forEach(Consumer { location: Location ->
                blocks[location] = borderMaterial
            })
            generatedBlockSchema.fillerBlocks.forEach(Consumer { location: Location ->
                blocks[location] = fillerMaterial
            })
            return BuildSchema(blocks)
        }
}